README.md

```diff
@@ -1,13 +1,208 @@
+production config after updating the local config? Have you ever been worried about
+the secrecy of your DB credentials and API keys by taking the `.env` file approach?
+Have you ever leaked confidential data by pushing secrets in your `.env` file
+to the git remote?
+
+If any of this has happened to you, **`strong-config` is here to help!**
+
+- Structure configs hierarchically by writing JSON or YAML.
+- Manage different configs for different environments.
+- Protect your secrets by applying strong cryptography. Encrypt secrets at rest
+  and automatically decrypt them at runtime by integrating a key management
+  service (KMS), e.g. [AWS KMS](https://aws.amazon.com/kms/) or
+  [Google Cloud KMS](https://cloud.google.com/kms/).
+- Enforce arbitrary permission models provided by your KMS. Decide who can encrypt
+  and decrypt configs for which environments.
+- Optionally validate them against a custom schema using
+  [`json-schema`](https://json-schema.org/). Also in CI.
+- Safeguard your configs by using git hooks. Make sure the configs are both
+  valid and encrypted before committing and pushing them.
+- Autogenerate Typescript types based on your schema.
+
+## Example
+
+Here is how an encrypted config file containing secrets looks like:
+
+```yml
+logger:
+  level: DEBUG
+auth:
+  url: my-auth-url
+  myApiKeySecret: ENC[AES256_GCM,data:aeQ+hlVIah7WyJoVR/Jbkb6GLH7ihsV0D81+U++pkiWD0zeoRL/Oe9Q3Tz6j/TNvKKVDnohIMyw3UVjELOuSY+A==,iv:nVRZWogV4B7o=,tag:KrE2jssfP4uCvqq+pc/JyQ==,type:str]
+sops:
+  gcp_kms:
+    - resource_id: projects/my-project/locations/europe-west3/keyRings/my-project-key-ring/cryptoKeys/my-strong-config-key
+      created_at: '2019-11-07T10:11:12Z'
+      enc: AiAAmdAgjOdw1XdV2MsDpvmZ4Deo867hmcX2B3DNhe2BCF2axuZ18hJJFK9oBlE1BrD70djwqi+L8T+NRNVnGUP+1//w8cJATAfJ8W/cQZFcdFTqjezC+VYv9xYI8i1bRna4xfFo/INIJtFDR38ZH1nrQg==
+  lastmodified: '2019-11-07T10:11:12Z'
+  mac: ENC[AES256_GCM,data:ABcd1EF2gh3IJKl4MNOpQr5stuvWXYz6sBCDEfGhIjK=,iv:A1AaAAAaa111a1Aa111AA/aaaAaaAAaa+aAaAaAAAaA=,tag:AAaaA1a1aaaAa/aa11AaaA==,type:str]
+  encrypted_suffix: Secret
+  version: 3.4.0
+```
+
+The field `sops` contains a set of metadata provided context information and
+important details to decrypt this config at runtime. Never delete this field
+or make changes to the encrypted file without using `@strong-config` - it will
+leave you with a corrupted config.
 
 ## How-to
 
-Set the environment variable `RUNTIME_ENVIRONMENT`
+Please backup your existing configs and secrets.
+
+1. Install `@strong-config/node` and the SOPS binary.
+
+   ```sh
+   npm install @strong-config/node
+   # or
+   yarn add @strong-config/node
+   ```
+
+   SOPS is [available for all major systems](https://github.com/mozilla/sops/releases).
+   If you are on Mac, you can install SOPS by running `brew install sops`.
+
+1. Define runtime environment and create first config file
+
+   ```sh
+   export NODE_ENV=development # or similar, depending on your shell
```

Comment:
To finish the discussion off from the previous thread: Still think this is confusing here.

Because you're assuming that the shell they're currently in will also be the one that they're going to call `yarn start` from. Which is not guaranteed.

I think what's missing is a last step in the how-to that's putting the NODE_ENV variable and the starting of the application into context:

```sh
# Important: NODE_DEV determines which file in the ./config folder will be loaded at runtime
NODE_DEV=development yarn start 
```

Reply:
Hm we have more assumptions

- people use the same shell or a sub-shell for all steps
- people use `yarn start` as a single entry point to the app
- people use bash or zsh

I just added a last step to the README. Let me know what you think. I believe we can't make everyone happy.

Reply:
Thoughts, @crosby ?

Reply:
Yeah, you can never make everyone happy. But javascript developers being familiar with `yarn start` is one of the safest assumptions to make in my book.

We should also not give examples for other shells than bash IMO, for the same reason. Bash is the default and everyone using zsh/fish/powershell etc will be familiar with translating bash commands to their respective shell.

if you remove the fish example from the new last step then: LGTM 👍

Reply:
Then we should change everything `yarn` to `npm` too right? https://www.npmtrends.com/yarn-vs-npm

Reply:
removed the fish line

Reply:
IMO in the end it's ok that `strong-config` also reflects _our opinions_ a little (which is currently preferring `yarn` over `npm`).

Plus, stats are tricky: https://trends.google.de/trends/explore?q=yarn,npm